File: Add Binary.cpp
class Solution { //Solution by Bans
public:
    string addBinary(string a, string b) {string ans="";int rem=0;
                                          int n1=a.length();int n2=b.length();
        int i=a.length()-1;int j=b.length()-1;int fi=0;int fj=0;
                                      
                                          
                                          
        while((i<n1 || j<n2 )||rem==1)
        { 
            
            if( ((a[i]-'0')+(b[j]-'0')+rem)==0)
              ans='0'+ans;
       else  if(((a[i]-'0')+(b[j]-'0')+rem)==1)
         { ans='1'+ans;rem=0;}
        else  if(((a[i]-'0')+(b[j]-'0')+rem)==2)
          {  ans='0'+ans;rem=1;}
         else if(((a[i]-'0')+(b[j]-'0')+rem)==3)
          {ans='1'+ans;rem=1;}
        if(i!=n1) i--;
            if(j!=n2)
            j--;
         if(i<0 && fi==0)
         {a=a+'0';i=a.length()-1;fi=1;}
         if(j<0 && fj==0)
         {b=b+'0';j=b.length()-1;fj=1;}
        }
        return ans;
        
    }
};

File: solutions/Add Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        na = len(a)
        nb = len(b)
        ans = ''
        carry = 0
        if na==nb:
            n = na
            for i in range(n-1, -1, -1):
                temp = int(a[i])+int(b[i])+carry
                carry = 0
                if temp==0:
                    ans = '0'+ans
                elif temp==1:
                    ans = '1'+ans
                elif temp==2:
                    ans = '0'+ans
                    carry = 1
                elif temp==3:
                    ans = '1'+ans
                    carry = 1
            if carry==0:
                return ans
            else:
                return '1'+ans
        if na>nb:
            major = a
            len_big = na
            minor = b
            len_sm = nb
        else:
            major = b
            len_big = nb
            minor = a
            len_sm = na
        for i in range(len_big-len_sm):
            minor = '0'+minor
        n = len_big
        for i in range(n-1, -1, -1):
            temp = int(major[i])+int(minor[i])+carry
            carry = 0
            if temp==0:
                ans = '0'+ans
            elif temp==1:
                ans = '1'+ans
            elif temp==2:
                ans = '0'+ans
                carry = 1
            elif temp==3:
                ans = '1'+ans
                carry = 1
        if carry==0:
            return ans
        else:
            return '1'+ans
